url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/io/filen/CashKyoroFile.java

```diff
@@ -0,0 +1,273 @@
+package net.hetimatan.io.filen;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.RandomAccessFile;
+import java.util.LinkedList;
+
+import net.hetimatan.io.file.KyoroByteOutput;
+import net.hetimatan.io.file.KyoroFile;
+import net.hetimatan.util.log.Log;
+
+//
+public class CashKyoroFile implements KyoroFile, KyoroByteOutput {
+	public static final String TAG ="RACashFile";
+	private RandomAccessFile mRAFile = null;
+	private File mFile = null;
+	private LinkedList<ByteKyoroFile> mCash = null;
+	private long mFilePointer = 0;
+	private long mLength = 0;
+	private int mChunkSize = 0;
+	private boolean mIsCash = false;
+	private ByteKyoroFile mCurrent = null;
+
+	public CashKyoroFile(byte[] data) throws IOException {
+		this(CashKyoroFileHelper.newCashFile(), data.length, 2);
+		addChunk(data);
+	}
+
+	public CashKyoroFile(int chunkSize) throws IOException {
+		this(CashKyoroFileHelper.newCashFile(), chunkSize, 2);
+		mIsCash = true;
+	}
+
+	public CashKyoroFile(int chunkSize, int chunkNum) throws IOException {
+		this(CashKyoroFileHelper.newCashFile(), chunkSize, chunkNum);
+		mIsCash = true;
+	}
+
+
+	public CashKyoroFile(File f, int chunkSize, int chunkNum) throws IOException {		
+		mChunkSize = chunkSize;
+		mFile = f;
+		if(mFile.exists()) {
+			mRAFile = new RandomAccessFile(mFile = f, "rw");
+		}
+		mIsCash = false;
+		mCash = new LinkedList<ByteKyoroFile>();
+		mLength = f.length();
+		for(int i=0;i<chunkNum;i++) {
+			ByteKyoroFile cash = new ByteKyoroFile(mChunkSize, mChunkSize);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *cash*. Currently, there are execution paths that do not contain closure statements, for example, when ByteKyoroFile.skip throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).